### ISSUE TITLE
Incorrect argument display for functions with arguments passed in xmm registers

### ISSUE THREAD

**Issue by Matir** [pwndbg/pwndbg#257]
### Description

pwndbg attempts to display the arguments that are being passed to a function call.  On x86-64, for functions with floating-point arguments passed in (xmm0, ...) the displayed arguments are still from rdi, etc.  One such function with this problem is `pow`.  (Details below).

### Steps to reproduce

Given the following source:
```
#include <stdio.h>
#include <math.h>

int main(int argc, char **argv) {
  double x = (double)argc + 2.0;
  return printf("%f\n", pow(10.0, x));
}
```

The context disasm view of pwndbg shows the following:

```
 ► 0x555555554762 <main+66>              call   pow@plt                       <0x5555555545d0>
        x: 0x1
        y: 0x7fffffffdab8 —▸ 0x7fffffffdf3c ◂— 0x6f6c662f706d742f ('/tmp/flo')
```

I expect x to have the value 10.0, and y to have the value 3.0.  (And those values are seen if the registers xmm0 and xmm1 are inspected, as per the x86-64 SysV ABI calling convention for doubles.)

### My setup

```
pwndbg> version
Gdb: GNU gdb (Debian 7.12-6) 7.12.0.20161007-git
Python: 3.5.3 (default, Jan 19 2017, 14:11:04)  [GCC 6.3.0 20170118]
Pwndbg: 1.0.0 build: 351d479f
```

**Comment by zachriggle** [comment]
It looks like the backing info, `functions.py`, does have correct floating-point arguments for these routines:

```
Function(type='double', derefcnt=0, name='pow', args=[Argument(type='double', derefcnt=0, name='x'), Argument(type='double', derefcnt=0, name='y')])
```

However, Pwndbg does not have any information on the floating-point ABI for any architecture:
https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/abi.py#L90-L113

If this is functionality you'd like to see, I recommend contributing a Pull Request that adds the informationto `abi.py`.

**Comment by Matir** [comment]
Fair enough, I'll try to figure out how to plumb that it.  Particularly complex is the case where a function has a mix of floating-point and integer/pointer arguments.  I don't know how hard it would be to put that together with the existing ABI API.

**Comment by disconnect3d** [comment]
Regarding this, shall we not show args in such cases or maybe display them with a warning they may not be accurate?

**Comment by aaryankumar1435-maker** [comment]
I’m interested in contributing to pwndbg and would like to work on this issue if it is still relevant. I’ve started setting up the project locally and am in the process of reproducing the behavior described here.

Please let me know if it would be okay for me to proceed.
